File: Python/gamex/store.py
@staticmethod
def getPathByKey(key):
    k,v = key.split(':', 2)
    match k:
        case 'Steam':
            from .store_steam import steamPaths
            return steamPaths[v] if v in steamPaths else None
        case 'GOG':
            from .store_gog import gogPaths
            return gogPaths[v] if v in gogPaths else None
        case 'Blizzard':
            from .store_blizzard import blizzardPaths
            return blizzardPaths[v] if v in blizzardPaths else None
        case 'Epic':
            from .store_epic import epicPaths
            return epicPaths[v] if v in epicPaths else None
        case 'Ubisoft':
            from .store_ubisoft import ubisoftPaths
            return ubisoftPaths[v] if v in ubisoftPaths else None
        case 'Unknown': return None
        case _: raise Exception(f'Unknown key: {key}')

# print(getPathByKey('Steam:1755910'))

File: dotnet/Base/GameX.Base/Store.cs
using GameX.Store;
using System;

namespace GameX
{
    /// <summary>
    /// StoreManager
    /// </summary>
    public static class StoreManager
    {
        public static string GetPathByKey(string key)
        {
            var parts = key.Split(':', 2);
            string k = parts[0], v = parts[1];
            return k switch
            {
                "Steam" => Store_Steam.SteamPaths.TryGetValue(v, out var z) ? z : null,
                "GOG" => Store_Gog.GogPaths.TryGetValue(v, out var z) ? z : null,
                "Blizzard" => Store_Blizzard.BlizzardPaths.TryGetValue(v, out var z) ? z : null,
                "Epic" => Store_Epic.EpicPaths.TryGetValue(v, out var z) ? z : null,
                "Ubisoft" => Store_Ubisoft.UbisoftPaths.TryGetValue(v, out var z) ? z : null,
                "Abandon" => Store_Abandon.AbandonPaths.TryGetValue(v, out var z) ? z : null,
                "Unknown" => null,
                _ => throw new ArgumentOutOfRangeException(nameof(key), key),
            };
        }
    }
}
